apollo-router/src/plugins/telemetry/config_new/graphql/mod.rs

```diff
@@ -0,0 +1,437 @@
+                .unwrap();
+
+            let harness = PluginTestHarness::<Telemetry>::builder()
+                .config(include_str!("fixtures/field_length_enabled.router.yaml"))
+                .schema(schema_str)
+                .build()
+                .await;
+
+            harness
+                .call_supergraph(request, |req| {
+                    let response: serde_json::Value = serde_json::from_str(include_str!(
+                        "../../../demand_control/cost_calculator/fixtures/federated_ships_named_response.json"
+                    ))
+                    .unwrap();
+                    supergraph::Response::builder()
+                        .data(response["data"].clone())
+                        .context(req.context)
+                        .build()
+                        .unwrap()
+                })
+                .await
+                .unwrap();
+
+            assert_histogram_sum!(
+                "graphql.field.list.length",
+                2.0,
+                "graphql.field.name" = "users",
+                "graphql.field.type" = "User",
+                "graphql.type.name" = "Query"
+            );
+        }
+        .with_metrics()
+        .await;
+    }
+
+    #[test_log::test(tokio::test)]
+    async fn multiple_fields_metric_publishing() {
+        async {
+            let schema_str = include_str!(
+                "../../../demand_control/cost_calculator/fixtures/federated_ships_schema.graphql"
+            );
+            let query_str = include_str!("../../../demand_control/cost_calculator/fixtures/federated_ships_fragment_query.graphql");
+
+
+            let request = supergraph::Request::fake_builder()
+                .query(query_str)
+                .context(context(schema_str, query_str))
+                .build()
+                .unwrap();
+
+            let harness = PluginTestHarness::<Telemetry>::builder()
+                .config(include_str!("fixtures/field_length_enabled.router.yaml"))
+                .schema(schema_str)
+                .build()
+                .await;
+
+            harness
+                .call_supergraph(request, |req| {
+                    let response: serde_json::Value = serde_json::from_str(include_str!(
+                        "../../../demand_control/cost_calculator/fixtures/federated_ships_fragment_response.json"
+                    ))
+                    .unwrap();
+                    supergraph::Response::builder()
+                        .data(response["data"].clone())
+                        .context(req.context)
+                        .build()
+                        .unwrap()
+                })
+                .await
+                .unwrap();
+
+            assert_histogram_sum!(
+                "graphql.field.list.length",
+                2.0,
+                "graphql.field.name" = "ships",
+                "graphql.field.type" = "Ship",
+                "graphql.type.name" = "Query" // TODO: I think this is wrong
```

Comment:
I left this note for myself, since it wasn't totally clear to me how `graphql.field.type` differed from `graphql.type.name`. Do you think folks configuring this feature will be able to tell these apart from the names/doc comments? Maybe something like `graphql.parent.type` would differentiate more?

```suggestion
                "graphql.type.name" = "Query"
```